kyo-zio/shared/src/main/scala/kyo/ZIOs.scala

```diff
@@ -55,47 +42,32 @@ object ZIOs:
       * @tparam A
       *   The result type
       */
-    inline def get[R: zio.Tag, E, A](v: ZIO[R, E, A])(using Tag[Env[R]], Frame): A < (Env[R] & ZIOs) =
+    inline def get[R: zio.Tag, E, A](v: ZIO[R, E, A])(using Tag[Env[R]], Frame): A < (Env[R] & Abort[E] & Async) =
         compiletime.error("ZIO environments are not supported yet. Please handle them before calling this method.")
 
-    /** Runs a Kyo effect within the ZIO runtime.
+    /** Interprets a Kyo computation to ZIO. Note that this method only accepts Abort[E] and Async pending effects. Plase handle any other
+      * effects before calling this method.
       *
       * @param v
       *   The Kyo effect to run
-      * @param frame
-      *   Implicit Frame
-      * @tparam E
-      *   The error type
-      * @tparam A
-      *   The result type
       * @return
-      *   A ZIO effect that will produce A or fail with E
+      *   A zio.ZIO that, when run, will execute the Kyo effect
       */
-    def run[E, A](v: => A < (Abort[E] & ZIOs))(using frame: Frame): ZIO[Any, E, A] =
+    def run[A: Flat, E](v: => A < (Abort[E] & Async))(using frame: Frame): ZIO[Any, E, A] =
         ZIO.suspendSucceed {
-            try
-                ArrowEffect.handle(Tag[GetZIO], v.map(r => ZIO.succeed(r): ZIO[Any, E, A]))(
-                    [C] => (input, cont) => input.flatMap(r => run(cont(r)).flatten)
-                ).pipe(Async.run).map { fiber =>
-                    ZIO.asyncInterrupt[Any, E, A] { cb =>
-                        fiber.unsafe.onComplete {
-                            case Result.Fail(ex)   => cb(Exit.fail(ex))
-                            case Result.Panic(ex)  => cb(Exit.die(ex))
-                            case Result.Success(v) => cb(v)
-                        }
-                        Left(ZIO.succeed {
-                            fiber.unsafe.interrupt(Result.Panic(Fiber.Interrupted(frame)))
-                        })
-                    }
-                }.pipe(IO.run).eval
-            catch
-                case ex =>
-                    ZIO.isFatalWith { isFatal =>
-                        if !isFatal(ex) then Exit.die(ex)
-                        else throw ex
+            Async.run(v).map { fiber =>
+                ZIO.asyncInterrupt[Any, E, A] { cb =>
+                    fiber.unsafe.onComplete {
+                        case Result.Success(a) => cb(ZIO.succeed(a))
```

Comment:
```suggestion
                        case Result.Success(a) => cb(Exit.succeed(a))
```
Suspension is not necessary here. `ZIO.die/fail` capture additional tracing information that `Exit.die/fail` do not. Not sure if that's necessary.